New_PlatformGame/Assets/NewPlayerMovement.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+public class NewPlayerMovement : MonoBehaviour
+{
+    //public 변수
+    public float jumpForceMax = 200f; //점프하는 힘의 최대치
+    public Sprite gotJumpBarSprite; //점프바의 스프라이트를 받기위한 변수
+    public float jumpChargeSpeed = 1f; //점프하는 힘을 더하는 속도
+    public float jumpBarLength; //점프바의 길이
+    public float xMoveSpeed = 5f; //움직임 속도
+    public float xMoveSpeedFly = 1f; //공중에 떠있을때 움직임 속도
+    public float jumpBarPositionX = 1f; //점프바의 위치 X (캐릭터 기준)
+    public float jumpBarPositionY = 1f; //점프바의 위치 Y (캐릭터 기준)
+
+    //private 변수
+    private Rigidbody2D _rigidbody2D;
+    private Collider2D _collider2D;
+    private GameObject _jumpBar; //점프바 변수
+    private SpriteRenderer _jumpBarSprite; //점프바의 스프라이트 렌더러
+    private float _xMove;
+    private Vector2 _getvel; //Move 함수에서 캐릭터의 속도를 받는 변수
+    private float _jumpForce; //점프하는 힘 변수
+    private bool _isCharge; //점프하는 힘을 모으고 있는걸 확인하는 변수
+    private bool _canJump; //땅에 있을때만 true
+    private float _direction; //점프한 방향
+    private bool _isAddJumpForceReverse; //점프하는 힘을 모으는 방향이 역방향인지 확인하는 변수
+
+    void Start()
+    {
+        _rigidbody2D = GetComponent<Rigidbody2D>();
+        _collider2D = GetComponent<Collider2D>();
+
+        //JumpBar 초기 설정
+        _jumpBar = new GameObject("Player JumpBar");
+        _jumpBar.AddComponent<SpriteRenderer>();
+        _jumpBarSprite = _jumpBar.GetComponent<SpriteRenderer>();
```

Comment:
```suggestion
        _jumpBarSprite = _jumpBar.GetComponent<SpriteRenderer>();
```
`GetComponent로 가져올 변수는 _(component type)로 통일` 하라는 코드 컨벤션이 있는데, 이 경우 어떤 오브젝트의 컴포넌트인지 상세히 알 수 없게 됩니다. 코드 컨벤션에 맞게 바꾸면 `_spriteRenderer` 라는 이름으로 고쳐야 하는데, 해당 코드 컨벤션을 수정하는 것을 추천합니다.